examples/2016-10-31/encryption_proxy/template.yaml

```diff
@@ -0,0 +1,167 @@
+                - "kms:CancelKeyDeletion"
+              Principal:
+                AWS: !Ref AWS::AccountId
+              Resource: '*'
+            - 
+              Sid: "Allow Encryption Service to use this key"
+              Effect: "Allow"
+              Action:
+                - "kms:Encrypt"
+              Principal:
+                AWS: !GetAtt EncryptionServiceIAMRole.Arn
+              Resource: '*'
+            - 
+              Sid: "Allow Decryption Service to use this key"
+              Effect: "Allow"
+              Action:
+                - "kms:Decrypt"
+              Principal:
+                AWS: !GetAtt DecryptionServiceIAMRole.Arn
+              Resource: '*'
+
+    EncryptionServiceIAMRole:
+        Type: "AWS::IAM::Role"
+        Properties:
+            Path: "/"
+            AssumeRolePolicyDocument:
+              Version: "2012-10-17"
+              Statement:
+                -
+                  Sid: "AllowLambdaServiceToAssumeRole"
+                  Effect: "Allow"
+                  Action: 
+                    - "sts:AssumeRole"
+                  Principal:
+                    Service: 
+                      - "lambda.amazonaws.com"
+            Policies:
+              -
+                PolicyName: "LambdaBasicExecution"
+                PolicyDocument:
+                  Version: '2012-10-17'
+                  Statement:
+                    -
+                      Sid: "AllowLambdaServiceToPushLogToCloudwatch"
+                      Effect: "Allow"
+                      Action:
+                        - "logs:CreateLogStream"
+                        - "logs:CreateLogGroup"
+                        - "logs:PutLogEvents"
+                      Resource: "arn:aws:logs:*:*:*"
+
+    DecryptionServiceIAMRole:
+        Type: "AWS::IAM::Role"
+        Properties:
+            Path: "/"
+            AssumeRolePolicyDocument:
+              Version: "2012-10-17"
+              Statement:
+                -
+                  Sid: "AllowLambdaServiceToAssumeRole"
+                  Effect: "Allow"
+                  Action: 
+                    - "sts:AssumeRole"
+                  Principal:
+                    Service: 
+                      - "lambda.amazonaws.com"
+            Policies:
+              -
+                PolicyName: "BasicExecution"
+                PolicyDocument:
+                  Version: '2012-10-17'
+                  Statement:
+                    -
+                      Sid: "AllowLambdaServiceToPushLogToCloudwatch"
+                      Effect: "Allow"
+                      Action:
+                        - "logs:CreateLogStream"
+                        - "logs:CreateLogGroup"
+                        - "logs:PutLogEvents"
+                      Resource: "arn:aws:logs:*:*:*"
+
+    EncryptionFunction:
+      Type: 'AWS::Serverless::Function'
+      Properties:
+        Handler: index.post
+        Runtime: python2.7
+        CodeUri: ./src/encryption
```

Comment:
To be consistent with other examples, set CodeUri to s3://<bucket-name>/encryption_proxy.zip